$OpenBSD: patch-util_oslib-posix_c,v 1.1 2018/04/23 09:02:26 ajacoutot Exp $
Pass MAP_STACK flag when allocating memory for stack usage.
Index: util/oslib-posix.c
--- util/oslib-posix.c.orig
+++ util/oslib-posix.c
@@ -538,7 +538,7 @@ void *qemu_alloc_stack(size_t *sz)
     *sz += pagesz;
 
     ptr = mmap(NULL, *sz, PROT_READ | PROT_WRITE,
-               MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+               MAP_PRIVATE | MAP_ANONYMOUS | MAP_STACK, -1, 0);
     if (ptr == MAP_FAILED) {
         perror("failed to allocate memory for stack");
         abort();
